include/seastar/core/internal/pollable_fd.hh

```diff
@@ -47,6 +49,15 @@ class packet;
 
 }
 
+/// Exception thrown when \ref pollable_fd object has been closed during active 
+/// waiting promise on \ref pollable_fd::readable or pollable_fd::writeable.
+class pollable_fd_aborted : public abort_requested_exception {
```

Comment:
Let's not mix in abort_source with the reactor. The reactor is the lowest level thing we have, abort_source is above it. So you can just inherit from std::runtime_exception.

Reply:
Do you mean std::runtime_error or std::exception? There is no std::runtime_exception in current C++ standard.

Reply:
replaced `seastar::abort_requested_exception` with std::exception.